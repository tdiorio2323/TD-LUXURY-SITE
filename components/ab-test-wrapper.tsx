"use client"

import { useEffect, useState } from 'react'
import { useABTest } from '@/lib/ab-testing'

interface ABTestWrapperProps {
  testId: string
  controlComponent: React.ReactNode
  variantComponent: React.ReactNode
  fallback?: React.ReactNode
}

export function ABTestWrapper({
  testId,
  controlComponent,
  variantComponent,
  fallback
}: ABTestWrapperProps) {
  const [mounted, setMounted] = useState(false)
  const { variant, isControl, isVariant, isInTest } = useABTest(testId)

  useEffect(() => {
    setMounted(true)
  }, [])

  // Prevent hydration mismatch by showing fallback until mounted
  if (!mounted) {
    return <>{fallback || controlComponent}</>
  }

  // If user is not in test, show fallback or control
  if (!isInTest) {
    return <>{fallback || controlComponent}</>
  }

  // Show appropriate variant
  if (isControl) {
    return <>{controlComponent}</>
  }

  if (isVariant) {
    return <>{variantComponent}</>
  }

  // Fallback
  return <>{fallback || controlComponent}</>
}

// Hook for conditional rendering based on A/B test
export function useABTestVariant(testId: string) {
  const [mounted, setMounted] = useState(false)
  const { variant, isControl, isVariant, isInTest, trackConversion } = useABTest(testId)

  useEffect(() => {
    setMounted(true)
  }, [])

  return {
    variant: mounted ? variant : null,
    isControl: mounted ? isControl : true, // Default to control during SSR
    isVariant: mounted ? isVariant : false,
    isInTest: mounted ? isInTest : false,
    trackConversion,
    mounted
  }
}