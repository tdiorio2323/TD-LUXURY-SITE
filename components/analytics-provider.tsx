"use client"

import { useEffect } from 'react'
import { usePathname } from 'next/navigation'
import { trackPageView, trackScrollDepth, trackTimeOnPage } from '@/lib/analytics'

export function AnalyticsProvider({ children }: { children: React.ReactNode }) {
  const pathname = usePathname()

  useEffect(() => {
    // Track page view on route change
    trackPageView(pathname)

    // Track scroll depth
    let maxScroll = 0
    const handleScroll = () => {
      const scrollPercent = Math.round((window.scrollY / (document.body.scrollHeight - window.innerHeight)) * 100)

      // Track scroll milestones (25%, 50%, 75%, 100%)
      if (scrollPercent >= 25 && maxScroll < 25) {
        trackScrollDepth(25)
        maxScroll = 25
      } else if (scrollPercent >= 50 && maxScroll < 50) {
        trackScrollDepth(50)
        maxScroll = 50
      } else if (scrollPercent >= 75 && maxScroll < 75) {
        trackScrollDepth(75)
        maxScroll = 75
      } else if (scrollPercent >= 100 && maxScroll < 100) {
        trackScrollDepth(100)
        maxScroll = 100
      }
    }

    // Track time on page
    const startTime = Date.now()
    const trackTime = () => {
      const timeSpent = Math.round((Date.now() - startTime) / 1000)
      trackTimeOnPage(timeSpent)
    }

    // Add event listeners
    window.addEventListener('scroll', handleScroll, { passive: true })
    window.addEventListener('beforeunload', trackTime)

    // Cleanup
    return () => {
      window.removeEventListener('scroll', handleScroll)
      window.removeEventListener('beforeunload', trackTime)
    }
  }, [pathname])

  return <>{children}</>
}