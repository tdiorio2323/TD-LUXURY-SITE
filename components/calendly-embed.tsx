"use client"

interface CalendlyEmbedProps {
  url: string
  height?: number
}

export function CalendlyEmbed({ url, height = 760 }: CalendlyEmbedProps) {
  return (
    <div className="relative w-full rounded-lg overflow-hidden" style={{ height: `${height}px` }}>
      <iframe
        src={url}
        width="100%"
        height="100%"
        frameBorder="0"
        title="Schedule a consultation with TD Studios"
        className="absolute inset-0"
      />
    </div>
  )
}
