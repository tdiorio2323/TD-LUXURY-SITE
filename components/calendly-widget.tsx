"use client"

import { InlineWidget } from "react-calendly"
import { useState, useEffect } from "react"

interface CalendlyWidgetProps {
  url: string
  className?: string
}

export function CalendlyWidget({ url, className = "" }: CalendlyWidgetProps) {
  const [failed, setFailed] = useState(false)
  const [loaded, setLoaded] = useState(false)

  useEffect(() => {
    // Set a timeout to detect if widget fails to load
    const timer = setTimeout(() => {
      if (!loaded) {
        setFailed(true)
      }
    }, 10000) // 10 second timeout

    return () => clearTimeout(timer)
  }, [loaded])

  if (failed) {
    return (
      <div className={`${className} p-8 bg-neutral-900/70 border border-white/20 rounded-lg text-center`}>
        <p className="text-white mb-4">Calendar failed to load.</p>
        <p className="text-white/80 text-sm">
          Please email us directly at{" "}
          <a className="underline text-white hover:text-white/80" href="mailto:tyler@tdstudiosny.com">
            tyler@tdstudiosny.com
          </a>{" "}
          or call{" "}
          <a className="underline text-white hover:text-white/80" href="tel:347-485-9935">
            347-485-9935
          </a>
        </p>
      </div>
    )
  }

  return (
    <div className={className}>
      <InlineWidget
        url={url}
        styles={{
          height: "700px",
          width: "100%",
        }}
        pageSettings={{
          backgroundColor: "0F0F10",
          hideEventTypeDetails: false,
          hideLandingPageDetails: false,
          primaryColor: "ffffff",
          textColor: "ffffff",
        }}
        onLoad={() => setLoaded(true)}
      />
    </div>
  )
}
