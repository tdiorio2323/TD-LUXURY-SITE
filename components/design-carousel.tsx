"use client"

import { useCallback, useEffect, useState } from 'react'
import useEmblaCarousel from 'embla-carousel-react'
import { ChevronLeft, ChevronRight } from 'lucide-react'
import Image from "next/image"

const portfolioImages = [
  {
    src: "/design-portfolio-1.jpg",
    alt: "Luxury brand design portfolio showcase",
    title: "Premium Brand Identity",
    description: "Sophisticated visual identity system"
  },
  {
    src: "/design-portfolio-2.jpg",
    alt: "Elegant website design and branding",
    title: "Digital Brand Experience",
    description: "Luxury web design and digital assets"
  },
  {
    src: "/design-portfolio-3.jpg",
    alt: "Creative design and typography showcase",
    title: "Typography & Layout",
    description: "Premium editorial design solutions"
  },
  {
    src: "/design-portfolio-4.jpg",
    alt: "Brand collateral and marketing materials",
    title: "Brand Collateral",
    description: "Complete marketing material suite"
  }
]

export function DesignCarousel() {
  const [emblaRef, emblaApi] = useEmblaCarousel({
    loop: true,
    align: 'center',
    skipSnaps: false,
    dragFree: false
  })
  const [selectedIndex, setSelectedIndex] = useState(0)
  const [scrollSnaps, setScrollSnaps] = useState<number[]>([])

  const scrollPrev = useCallback(() => {
    if (emblaApi) emblaApi.scrollPrev()
  }, [emblaApi])

  const scrollNext = useCallback(() => {
    if (emblaApi) emblaApi.scrollNext()
  }, [emblaApi])

  const scrollTo = useCallback((index: number) => {
    if (emblaApi) emblaApi.scrollTo(index)
  }, [emblaApi])

  const onInit = useCallback((emblaApi: any) => {
    setScrollSnaps(emblaApi.scrollSnapList())
  }, [])

  const onSelect = useCallback((emblaApi: any) => {
    setSelectedIndex(emblaApi.selectedScrollSnap())
  }, [])

  useEffect(() => {
    if (!emblaApi) return

    onInit(emblaApi)
    onSelect(emblaApi)
    emblaApi.on('reInit', onInit)
    emblaApi.on('select', onSelect)
  }, [emblaApi, onInit, onSelect])

  // Auto-play functionality
  useEffect(() => {
    if (!emblaApi) return

    const autoplay = setInterval(() => {
      emblaApi.scrollNext()
    }, 4000) // Change slide every 4 seconds

    return () => clearInterval(autoplay)
  }, [emblaApi])

  return (
    <div className="relative w-full max-w-6xl mx-auto">
      <div className="overflow-hidden rounded-lg" ref={emblaRef}>
        <div className="flex">
          {portfolioImages.map((image, index) => (
            <div
              key={index}
              className="flex-[0_0_100%] md:flex-[0_0_50%] lg:flex-[0_0_33.333%] min-w-0 px-2"
            >
              <div className="relative group">
                <div className="aspect-[4/3] overflow-hidden rounded-lg bg-gray-800">
                  <Image
                    src={image.src}
                    alt={image.alt}
                    fill
                    style={{ objectFit: "cover" }}
                    sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
                    className="transition-transform duration-500 group-hover:scale-105"
                  />
                  {/* Overlay */}
                  <div className="absolute inset-0 bg-black/40 opacity-0 group-hover:opacity-100 transition-opacity duration-300 flex items-end">
                    <div className="p-6 text-white">
                      <h4 className="text-lg font-semibold mb-2">{image.title}</h4>
                      <p className="text-sm text-white/80">{image.description}</p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>

      {/* Navigation Arrows */}
      <button
        className="absolute left-4 top-1/2 -translate-y-1/2 p-3 bg-black/50 backdrop-blur-sm rounded-full text-white hover:bg-black/70 transition-all duration-300 focus:outline-none focus:ring-2 focus:ring-white/30 mobile-touch-target min-h-[44px] px-4 py-3"
        onClick={scrollPrev}
        aria-label="Previous image"
      >
        <ChevronLeft className="w-6 h-6" />
      </button>

      <button
        className="absolute right-4 top-1/2 -translate-y-1/2 p-3 bg-black/50 backdrop-blur-sm rounded-full text-white hover:bg-black/70 transition-all duration-300 focus:outline-none focus:ring-2 focus:ring-white/30 mobile-touch-target min-h-[44px] px-4 py-3"
        onClick={scrollNext}
        aria-label="Next image"
      >
        <ChevronRight className="w-6 h-6" />
      </button>

      {/* Dots Indicator */}
      <div className="flex justify-center space-x-2 mt-6">
        {scrollSnaps.map((_, index) => (
          <button
            key={index}
            className={`w-3 h-3 rounded-full transition-all duration-300 focus:outline-none focus:ring-2 focus:ring-white/30 ${
              index === selectedIndex
                ? 'bg-white'
                : 'bg-white/30 hover:bg-white/50'
            }`}
            onClick={() => scrollTo(index)}
            aria-label={`Go to slide ${index + 1}`}
          />
        ))}
      </div>
    </div>
  )
}