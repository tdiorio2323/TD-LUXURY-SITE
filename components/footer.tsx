import Link from "next/link"

export function Footer() {
  return (
    <footer className="relative border-t border-white/10 overflow-hidden">
      {/* Background Image with Overlay */}
      <div className="absolute inset-0">
        <img
          src="/footer-background.jpg"
          alt="Footer Background"
          className="w-full h-full object-cover"
        />
        <div className="absolute inset-0 bg-black/10"></div>
      </div>

      <div className="relative z-10 max-w-7xl mx-auto px-4 sm:px-6 py-12 sm:py-16">
        <div className="text-center mb-6 sm:mb-8">
          <h2 className="text-xl sm:text-2xl md:text-3xl font-bold text-white mb-3 sm:mb-4">Let's build the next flagship experience.</h2>
          <p className="text-white max-w-2xl mx-auto mb-6 sm:mb-8 text-sm sm:text-base">
            Strategy, design, development, and social programs for ambitious brands headquartered in New York City,
            serving teams worldwide.
          </p>
          <Link
            href="/contact"
            className="inline-flex items-center justify-center px-6 sm:px-8 py-2 sm:py-3 bg-neutral-900/70 backdrop-blur-sm border border-white/20 rounded-lg text-white font-medium hover:bg-neutral-900/80 transition-colors text-sm sm:text-base"
          >
            Start your project
          </Link>
        </div>

        <div className="text-center text-white text-sm">Â© 2025 TD Studios. All rights reserved.</div>
      </div>
    </footer>
  )
}
