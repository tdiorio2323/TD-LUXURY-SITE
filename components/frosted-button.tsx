"use client"

import Link from "next/link"
import type { ReactNode } from "react"
import { trackButtonClick, trackCTAPerformance } from "@/lib/analytics"

interface FrostedButtonProps {
  href?: string
  children: ReactNode
  className?: string
  onClick?: () => void
  type?: "button" | "submit" | "reset"
  analyticsLabel?: string // Optional label for analytics tracking
  analyticsPosition?: string // Optional position for analytics
}

export function FrostedButton({ href, children, className = "", onClick, type = "button", analyticsLabel, analyticsPosition }: FrostedButtonProps) {
  const baseClasses =
    "inline-flex items-center justify-center px-6 py-3 bg-neutral-900/70 backdrop-blur-sm border border-white/20 rounded-lg text-white font-medium hover:bg-neutral-900/80 transition-all duration-300 mobile-touch-target mobile-btn-primary md:mobile-btn-primary"

  const handleClick = () => {
    const buttonText = typeof children === 'string' ? children : 'Button'
    const label = analyticsLabel || buttonText
    const position = analyticsPosition || 'unknown'

    trackButtonClick(label, position)
    trackCTAPerformance(label, position)

    if (onClick) {
      onClick()
    }
  }

  if (href) {
    return (
      <Link
        href={href}
        className={`${baseClasses} ${className}`}
        onClick={handleClick}
      >
        {children}
      </Link>
    )
  }

  return (
    <button type={type} onClick={handleClick} className={`${baseClasses} ${className}`}>
      {children}
    </button>
  )
}
