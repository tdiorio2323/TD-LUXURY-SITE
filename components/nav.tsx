"use client"

import { useState } from "react"
import Link from "next/link"
import { usePathname } from "next/navigation"
import { Logo } from "./logo"
import { Instagram, Send, Menu, X } from "lucide-react"

const navItems = [
  { name: "HOME", href: "/" },
  { name: "WEB", href: "/web" },
  { name: "DEV", href: "/dev" },
  { name: "SOCIAL", href: "/social" },
  { name: "RESOURCES", href: "/resources" },
  { name: "PORTFOLIO", href: "/portfolio" },
  { name: "CONTACT", href: "/contact" },
]

export function Nav() {
  const pathname = usePathname()
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false)

  const toggleMobileMenu = () => {
    setIsMobileMenuOpen(!isMobileMenuOpen)
  }

  return (
    <nav className="fixed top-0 left-0 right-0 z-50 bg-black/80 backdrop-blur-sm border-b border-white/10">
      <div className="max-w-7xl mx-auto px-6 py-8">
        <div className="flex items-center justify-between">
          <Link href="/" className="flex items-center" aria-label="TD Studios - Home">
            <Logo className="h-16" aria-hidden="true" />
            <span className="sr-only">TD Studios</span>
          </Link>

          {/* Desktop Navigation */}
          <div className="hidden md:flex items-center space-x-12">
            {navItems.map((item) => (
              <Link
                key={item.name}
                href={item.href}
                className={`text-lg font-medium tracking-wider transition-colors hover:text-white ${
                  pathname === item.href ? "text-white border-b border-white" : "text-white/70"
                }`}
              >
                {item.name}
              </Link>
            ))}
          </div>

          {/* Social Icons and Mobile Menu */}
          <div className="flex items-center space-x-4">
            <a
              href="https://instagram.com/tdstudiosco"
              target="_blank"
              rel="noopener noreferrer"
              aria-label="Follow TD Studios on Instagram"
              className="p-4 bg-neutral-900/70 backdrop-blur-sm rounded-lg hover:bg-neutral-900/80 transition-all duration-300 group focus:outline-none focus:ring-2 focus:ring-white/30"
            >
              <Instagram className="w-6 h-6 text-white group-hover:scale-110 transition-transform" aria-hidden="true" />
            </a>
            <a
              href="https://t.me/tdstudioscorp"
              target="_blank"
              rel="noopener noreferrer"
              aria-label="Connect with TD Studios on Telegram"
              className="p-4 bg-neutral-900/70 backdrop-blur-sm rounded-lg hover:bg-neutral-900/80 transition-all duration-300 group focus:outline-none focus:ring-2 focus:ring-white/30"
            >
              <Send className="w-6 h-6 text-white group-hover:scale-110 transition-transform" aria-hidden="true" />
            </a>

            {/* Mobile Menu Button */}
            <button
              onClick={toggleMobileMenu}
              aria-label={isMobileMenuOpen ? "Close mobile menu" : "Open mobile menu"}
              aria-expanded={isMobileMenuOpen}
              className="md:hidden p-4 bg-neutral-900/70 backdrop-blur-sm rounded-lg hover:bg-neutral-900/80 transition-all duration-300 focus:outline-none focus:ring-2 focus:ring-white/30 mobile-touch-target"
            >
              {isMobileMenuOpen ? (
                <X className="w-6 h-6 text-white" aria-hidden="true" />
              ) : (
                <Menu className="w-6 h-6 text-white" aria-hidden="true" />
              )}
            </button>
          </div>
        </div>
      </div>

      {/* Mobile Menu Overlay */}
      {isMobileMenuOpen && (
        <div className="md:hidden fixed inset-0 top-32 bg-black z-40">
          <div className="flex flex-col items-center justify-start pt-16 min-h-[60vh] space-y-2 px-6">
            {navItems.map((item) => (
              <Link
                key={item.name}
                href={item.href}
                onClick={() => setIsMobileMenuOpen(false)}
                className={`w-full text-center text-2xl font-medium tracking-wider transition-colors hover:text-white mobile-nav-item mobile-touch-target flex items-center justify-center ${
                  pathname === item.href ? "text-white border-b-2 border-white pb-2" : "text-white/70"
                }`}
              >
                {item.name}
              </Link>
            ))}
          </div>
        </div>
      )}
    </nav>
  )
}
