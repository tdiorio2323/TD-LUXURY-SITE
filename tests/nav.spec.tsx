import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import React from "react";
import { Nav } from "../components/nav";

// Mock window.location
Object.defineProperty(window, "location", {
    value: {
        host: "localhost:3000",
    },
    writable: true,
});

describe("Nav", () => {
    it("renders stable snapshot", () => {
        const { asFragment } = render(<Nav />);
        expect(asFragment()).toMatchSnapshot();
    });
});