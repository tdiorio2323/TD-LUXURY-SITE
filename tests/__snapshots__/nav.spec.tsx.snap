// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav > renders stable snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-black/80 backdrop-blur-sm border-b border-white/10"
  >
    <div
      class="hidden md:block"
    >
      <div
        class="max-w-7xl mx-auto px-6 pt-6 pb-2"
      >
        <div
          class="flex justify-center"
        >
          <a
            aria-label="TD Studios - Home"
            class="flex items-center"
            href="/"
          >
            <div
              class="select-none h-16"
            >
              <img
                alt="TD Studios Logo"
                class="h-16 w-auto"
                height="64"
                src="/images/td-studios-logo.png"
                width="256"
              />
            </div>
            <span
              class="sr-only"
            >
              TD Studios
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="max-w-7xl mx-auto px-6 py-4 md:py-4"
    >
      <div
        class="flex items-center justify-between"
      >
        <div
          class="md:hidden flex-1 flex justify-center"
        >
          <a
            aria-label="TD Studios - Home"
            class="flex items-center"
            href="/"
          >
            <div
              class="select-none h-12"
            >
              <img
                alt="TD Studios Logo"
                class="h-16 w-auto"
                height="64"
                src="/images/td-studios-logo.png"
                width="256"
              />
            </div>
            <span
              class="sr-only"
            >
              TD Studios
            </span>
          </a>
        </div>
        <div
          class="hidden md:flex items-center justify-center flex-1 space-x-12"
        >
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/work"
          >
            WORK
          </a>
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/web"
          >
            WEB
          </a>
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/dev"
          >
            DEV
          </a>
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/social"
          >
            SOCIAL
          </a>
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/design"
          >
            DESIGN
          </a>
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/process"
          >
            PROCESS
          </a>
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/resources"
          >
            RESOURCES
          </a>
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/faq"
          >
            FAQ
          </a>
          <a
            class="text-lg font-medium tracking-wider transition-colors hover:text-white text-white/70"
            href="/contact"
          >
            CONTACT
          </a>
        </div>
        <div
          class="hidden md:flex items-center space-x-4"
        >
          <a
            aria-label="Call TD Studios at 347-485-9935"
            class="p-4 bg-neutral-900/70 backdrop-blur-sm rounded-lg hover:bg-blue-500/20 transition-all duration-300 group focus:outline-none focus:ring-2 focus:ring-blue-500/50"
            href="tel:347-485-9935"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-send w-6 h-6 text-white group-hover:text-blue-500 transition-colors"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.536 21.686a.5.5 0 0 0 .937-.024l6.5-19a.496.496 0 0 0-.635-.635l-19 6.5a.5.5 0 0 0-.024.937l7.93 3.18a2 2 0 0 1 1.112 1.11z"
              />
              <path
                d="m21.854 2.147-10.94 10.939"
              />
            </svg>
          </a>
          <a
            aria-label="Message TD Studios on WhatsApp"
            class="p-4 bg-neutral-900/70 backdrop-blur-sm rounded-lg hover:bg-green-500/20 transition-all duration-300 group focus:outline-none focus:ring-2 focus:ring-green-500/50"
            href="https://wa.me/13474859935"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-message-circle w-6 h-6 text-white group-hover:text-green-500 transition-colors"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
              />
            </svg>
          </a>
          <a
            aria-label="Follow TD Studios on Instagram"
            class="p-4 bg-neutral-900/70 backdrop-blur-sm rounded-lg hover:bg-neutral-900/80 transition-all duration-300 group focus:outline-none focus:ring-2 focus:ring-white/30"
            href="https://instagram.com/tdstudiosco"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-instagram w-6 h-6 text-white group-hover:scale-110 transition-transform"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="20"
                rx="5"
                ry="5"
                width="20"
                x="2"
                y="2"
              />
              <path
                d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
              />
              <line
                x1="17.5"
                x2="17.51"
                y1="6.5"
                y2="6.5"
              />
            </svg>
          </a>
        </div>
        <button
          aria-expanded="false"
          aria-label="Open mobile menu"
          class="md:hidden p-4 bg-neutral-900/70 backdrop-blur-sm rounded-lg hover:bg-neutral-900/80 transition-all duration-300 focus:outline-none focus:ring-2 focus:ring-white/30 mobile-touch-target"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-menu w-6 h-6 text-white"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="20"
              y1="12"
              y2="12"
            />
            <line
              x1="4"
              x2="20"
              y1="6"
              y2="6"
            />
            <line
              x1="4"
              x2="20"
              y1="18"
              y2="18"
            />
          </svg>
        </button>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
